packages/process/process-monitor.pony

```diff
@@ -0,0 +1,366 @@
+use "files"
+
+use @pony_os_errno[I32]()
+use @pony_asio_event_create[AsioEventID](owner: AsioEventNotify, fd: U32, 
+      flags: U32, nsec: U64, noisy: Bool)
+use @pony_asio_event_unsubscribe[None](event: AsioEventID)
+use @pony_asio_event_destroy[None](event: AsioEventID)
+      
+primitive _EINTR        fun apply(): I32 => 4
+primitive _EAGAIN       fun apply(): I32 => 35
+primitive _STDINFILENO  fun apply(): U32 => 0
+primitive _STDOUTFILENO fun apply(): U32 => 1
+primitive _STDERRFILENO fun apply(): U32 => 2
+primitive _FSETFL       fun apply(): I32 => 4
+primitive _ONONBLOCK    fun apply(): I32 => 4
+
+primitive ExecveError
+primitive PipeError
+primitive Dup2Error
+primitive ForkError
+primitive FcntlError
+primitive WaitpidError
+primitive CloseError
+primitive ReadError
+primitive WriteError
+primitive Unsupported // we throw this on non POSIX systems
+
+type ProcessError is
+  ( ExecveError
+  | PipeError    
+  | Dup2Error    
+  | ForkError
+  | FcntlError
+  | WaitpidError
+  | CloseError
+  | ReadError
+  | WriteError
+  | Unsupported
+  )    
+  
+actor ProcessMonitor
+  """
+  Forks and monitors a process. Notifies a client about STDOUT / STDERR events.
+  """        
+  let _notifier: ProcessNotify
+  var _child_pid: I32 = -1
+  
+  var _stdout_event: AsioEventID = AsioEvent.none()
+  var _stderr_event: AsioEventID = AsioEvent.none()
+
+  var _read_buf: Array[U8] iso = recover Array[U8].undefined(4096) end
+
+  var _stdin_read:   U32 = -1
+  var _stdin_write:  U32 = -1
+  var _stdout_read:  U32 = -1
+  var _stdout_write: U32 = -1
+  var _stderr_read:  U32 = -1
+  var _stderr_write: U32 = -1
+    
+  var _stdout_open: Bool = true
+  var _stderr_open: Bool = true
+
+  var _closed: Bool = false
+  
+  new create(notifier: ProcessNotify iso, path: String, 
+    args: Array[String] val, vars: Array[String] val) =>
```

Comment:
the standard at this point would be to put the => on a newline indented to the same level as `new`.

that applies to all multiline method defintions.